### PR TITLE
core: move metric savings to audit product

### PR DESCRIPTION
**Pull request by adamraine**
Original approach was flawed (see https://github.com/GoogleChrome/lighthouse/pull/15071)

### CODE REVIEW

File: proto/lighthouse-result.proto [hunk 1]
```diff
@@ -377,6 +377,10 @@ message AuditResult {
   // The unit of the numeric_value field. Used to format the numeric value for
   // display.
   string numeric_unit = 12;
+
+  // [EXPERIMENTAL] Estimates of how much this audit affects various performance
+  // metrics. Values will be in the unit of the respective metrics.
+  google.protobuf.Struct _metricSavings = 13;
```
Comment (adamraine) [inline proto/lighthouse-result.proto:383]: Is renaming this to `metricSavings` going to cause issues down the line?

**Comment by brendankenny** [comment]
I didn't realize in https://github.com/GoogleChrome/lighthouse/pull/15071#discussion_r1192540908 that it wasn't on the actual audit result in the LHR, just on the intermediate audit product. Maybe the current way is the best way to land these, since we don't have to worry about them exposed in the LHR at all yet?

File: proto/lighthouse-result.proto (hunk 1 above)
Comment (brendankenny) [reply proto/lighthouse-result.proto:383]: (otherwise it's my understanding that it'll have to ship in the proto forever, just not be populated anymore)

**Comment by adamraine** [comment]
Protobuf seems to be having issues with the leading `_` :/

We may need to remove it.

**Review by connorjclark — APPROVED** [review]

**Pull request merged**